### ISSUE TITLE
Debugging reasons for truck re-route

### ISSUE THREAD

**Issue by kochis** [valhalla/valhalla#4697]
I'm seeing an instance where truck routes are being re-routed off the main highway, and trying to understand why.

Origin: `42.5525176,-114.5271132`
Destination: `46.32539,-119.30359`

Route: https://valhalla.openstreetmap.de/directions?profile=truck&wps=-114.4603729248047%2C42.563448700516524%2C-119.27925109863283%2C46.27958036616893
<img width="1633" alt="image" src="https://github.com/valhalla/valhalla/assets/814934/91ce2c36-e68e-42b7-8be9-625815197a83">

I would expect it to stay on the main highway, as the car route does:
<img width="1614" alt="image" src="https://github.com/valhalla/valhalla/assets/814934/7d878231-5c70-466d-9cc1-8a537d59579e">

I've tried almost every configuration of costing options that might be causing this, and it never seems to take that route, so I'm assuming it's something with the data. Interestingly, it _does_ use the highway in the other direction.

I've basically narrowed it down to something in this relation (12139901) https://www.openstreetmap.org/relation/12139901. 

I've also pulled out any of the nodes with tags here. [nodes.json](https://github.com/valhalla/valhalla/files/15042997/nodes.json)

I'm not too familiar with how the data is being leveraged in the routing, so hoping someone can help me understand how to figure out the case for the issue. Thanks!

**Comment by nilsnolde** [comment]
Can you try the same query but with „hgv_access_penalty: 0.0“ in truck costing options? If that works, then there’s somewhere a hgv=no edge.

**Comment by kochis** [comment]
It looks like this section has `hgv: no`

https://www.openstreetmap.org/way/684871963

**Comment by kochis** [comment]
Also yes, setting `hgv_access_penalty = 0.0` used the correct route. My hunch is that road segment is incorrectly tagged.

**Comment by kochis** [comment]
Setting `hgv_access_penalty = 0.0` did not change the route.

This how I am setting it (to make sure I'm doing it correctly)
```
/route?json={"locations":[{"lat":42.5525176,"lon":-114.5271132},{"lat":46.32539,"lon":-119.30359}],"costing":"truck","costing_options":{"truck":{"hgv_no_access_penalty":0}}}
```

**Comment by kochis** [comment]
I guess this change is not in the latest release? https://github.com/valhalla/valhalla/pull/4650

**Comment by nilsnolde** [comment]
Nope, we’re always assuming people test with master. We should put that in the issue template..

So it works with master right?

**Comment by kochis** [comment]
> So it works with master right?
 
I'll have to throw a build together and get back to you 😅 

But assuming that's the case, it seems like the `hgb: no` on this route segment is causing the issue: https://www.openstreetmap.org/way/684871963
